app/RouteHandlers/AddDonationHandler.php

```diff
@@ -183,4 +186,33 @@ private function isSubmissionAllowed( Request $request ) {
 		return true;
 	}
 
+	private function collectPiwikTrackingEvents( array $sessionTrackingData, Donation $donation ) {
+		$piwikTracking = new PiwikCustomTracking();
+		if ( isset( $sessionTrackingData['paymentType'] ) ) {
+			$piwikTracking->setCustomVariable(
+				1,
```

Comment:
Where do these IDs come from? I'm guessing something will break if they get changed around here. If so, some indication of what that is, and how to verify correctness here would be nice. As far as I can tell no tests will break if they are changed here (just tried, and indeed, they don't). Perhaps this is a case where adding such tests makes little sense. However the combination of not realizing there is something that can be messed up and no detection of that happening short of the fun people coming into the room shouting at you is not good.